dashboard/src/components/settings/roles/CreateRoleForm/CreateRoleForm.tsx

```diff
@@ -61,26 +66,38 @@ export default function CreateRoleForm({
       return;
     }
 
-    const updateAppPromise = updateApp({
+    const updateConfigPromise = updateConfig({
       variables: {
-        id: currentApplication?.id,
-        app: {
-          authUserDefaultAllowedRoles: `${data?.app?.authUserDefaultAllowedRoles},${name}`,
+        appId: currentApplication?.id,
+        config: {
+          auth: {
+            user: {
+              roles: {
+                allowed: [...(allowedRoles || []), name],
```

Comment:
low readability.

Maybe better to seperatly check that `allowedRoles` is an array and then just `allowedRoles.push(name)`?

Reply:
Could you please elaborate on the readability part?

array.push is an imperative function which mutates the source array. I'd rather not use it anywhere.

Reply:
> Could you please elaborate on the readability part?

I had to think more than 10 seconds to understand the code :D

Feel free to skip.

Reply:
I changed it to:

```ts
const updatedAllowedRoles = allowedRoles ? [...allowedRoles, name] : [name];

// ...
roles: {
  allowed: updatedAllowedRoles,
},
// ...
```